lib/anonymous.js

```diff
@@ -0,0 +1,214 @@
+import InvoiceStatus from '../components/invoice-status'
+import { sleep } from './time'
+import { Button } from 'react-bootstrap'
+import { CopyInput } from '../components/form'
+
+const Contacts = ({ invoiceHash }) => {
+  const subject = `Support request for payment hash: ${invoiceHash}`
+  const body = 'Hi, I successfully paid for <insert action> but the action did not work.'
+  return (
+    <div className='d-flex flex-column justify-content-center'>
+      <span>Payment hash</span>
+      <div className='w-100'>
+        <CopyInput type='text' placeholder={invoiceHash} readOnly noForm />
+      </div>
+      <div className='d-flex flex-row justify-content-center'>
+        <a
+          href={`mailto:kk@stacker.news?subject=${subject}&body=${body}`} className='nav-link p-0 d-inline-flex'
+          target='_blank' rel='noreferrer'
+        >
+          e-mail
+        </a>
+        <span className='mx-2 text-muted'> \ </span>
+        <a
+          href='https://tribes.sphinx.chat/t/stackerzchat' className='nav-link p-0 d-inline-flex'
+          target='_blank' rel='noreferrer'
+        >
+          sphinx
+        </a>
+        <span className='mx-2 text-muted'> \ </span>
+        <a
+          href='https://t.me/k00bideh' className='nav-link p-0 d-inline-flex'
+          target='_blank' rel='noreferrer'
+        >
+          telegram
+        </a>
+        <span className='mx-2 text-muted'> \ </span>
+        <a
+          href='https://simplex.chat/contact#/?v=1-2&smp=smp%3A%2F%2F6iIcWT_dF2zN_w5xzZEY7HI2Prbh3ldP07YTyDexPjE%3D%40smp10.simplex.im%2FebLYaEFGjsD3uK4fpE326c5QI1RZSxau%23%2F%3Fv%3D1-2%26dh%3DMCowBQYDK2VuAyEAV086Oj5yCsavWzIbRMCVuF6jq793Tt__rWvCec__viI%253D%26srv%3Drb2pbttocvnbrngnwziclp2f4ckjq65kebafws6g4hy22cdaiv5dwjqd.onion&data=%7B%22type%22%3A%22group%22%2C%22groupLinkId%22%3A%22cZwSGoQhyOUulzp7rwCdWQ%3D%3D%22%7D' className='nav-link p-0 d-inline-flex'
+          target='_blank' rel='noreferrer'
+        >
+          simplex
+        </a>
+      </div>
+    </div>
+  )
+}
+
+const Invoice = ({ id, hash, errorCount, repeat, ...props }) => {
+  const { data, loading, error } = useQuery(INVOICE, {
+    pollInterval: 1000,
+    variables: { id }
+  })
+  if (error) {
+    if (error.message?.includes('invoice not found')) {
+      return
+    }
+    return <div>error</div>
+  }
+  if (!data || loading) {
+    return <QrSkeleton status='loading' />
+  }
+
+  let errorStatus = 'Something went wrong. Please try again.'
+  if (errorCount > 1) {
+    errorStatus = 'Something still went wrong.\nPlease contact admins for support or to request a refund.'
+  }
+  return (
+    <>
+      <QrInvoice invoice={data.invoice} {...props} />
+      {errorCount > 0
+        ? (
+          <>
+            <InvoiceStatus variant='failed' status={errorStatus} />
+            {errorCount === 1
+              ? <div className='d-flex flex-row mt-3 justify-content-center'><Button variant='info' onClick={repeat}>Retry</Button></div>
```

Comment:
Not sure if this retry button is very useful. As it is, it is very naive and just literally retries the action, hoping it would succeed now.

We could try to parse the error and check if there is something that can be done. For example, if the error is insufficient fees, we could update the invoice amount :thinking:

Reply:
I think the retry button is better than nothing and its sufficient to ship it